src/consensus/cfund.cpp

```diff
@@ -898,122 +1008,115 @@ void CFund::CFundStep(const CValidationState& state, CBlockIndex *pindexNew, con
                 continue;
             }
 
-            if(fUndo && proposal->blockhash == pindexDelete->GetBlockHash())
-            {
-                proposal->blockhash = uint256();
-                proposal->fState = CFund::NIL;
-                fUpdate = true;
-            }
-
             CBlockIndex* pblockindex = mapBlockIndex[proposal->txblockhash];
 
+            bool fUpdate = false;
+
             int nCreatedOnCycle = (pblockindex->nHeight / Params().GetConsensus().nBlocksPerVotingCycle);
             int nCurrentCycle = (pindexNew->nHeight / Params().GetConsensus().nBlocksPerVotingCycle);
             int nElapsedCycles = std::max(nCurrentCycle - nCreatedOnCycle, 0);
             int nVotingCycles = std::min(nElapsedCycles, (int)Params().GetConsensus().nCyclesProposalVoting + 1);
 
-            auto oldState = proposal->fState;
             auto oldCycle = proposal->nVotingCycle;
 
-            if((proposal->fState == CFund::NIL || fUndo) && nVotingCycles != proposal->nVotingCycle)
+            CProposal tmp;
+            CProposal oldproposal = CProposal();
+
+            if (fLog)
```

Comment:
why was this also moved further down?

Reply:
CFundStep has been rewritten a bit to accomodate the changes of this pull request. this moved some parts of the code